pkg/egressgateway/manager.go

```diff
@@ -190,6 +198,35 @@ func (manager *Manager) runReconciliationAfterK8sSync(ctx context.Context) {
 	}()
 }
 
+// retryEndpointIdentityLabelsResolution spawns a goroutine that retries to
+// add/update an endpoint for which we failed to resolve the identity to a set
+// of labels
+func (manager *Manager) retryEndpointIdentityLabelsResolution(ctx context.Context) {
+	go func() {
+		ticker := time.NewTicker(1 * time.Second)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.C:
+				manager.Lock()
+				queueLen := manager.endpointEventRetryQueue.Len()
+				manager.Unlock()
+
+				if queueLen == 0 {
+					continue
+				}
+
+				endpointEventRetry := heap.Pop(&manager.endpointEventRetryQueue).(*endpointEventRetry)
+				endpointEventRetry.backoff.Wait(ctx)
+				manager.addEndpoint(endpointEventRetry.endpoint, endpointEventRetry.backoff)
```

Comment:
There's the possibility that an endpoint which is added here after the wait period has already been deleted in parallel.